internal/handlers/common/sort.go

```diff
@@ -40,6 +41,14 @@ func SortDocuments(docs []*types.Document, sortDoc *types.Document) error {
 	sortFuncs := make([]sortFunc, len(sortDoc.Keys()))
 
 	for i, sortKey := range sortDoc.Keys() {
+		if strings.Contains(sortKey, "$") {
```

Comment:
```suggestion
		if strings.HasPrefix(sortKey, "$") {
```

I suggest `HasPrefix` for clarity (the error is about `$` being the first symbol), the rest looks good to me.